security/core/security-core-impl/src/main/java/ddf/security/assertion/impl/SecurityAssertionImpl.java

```diff
@@ -270,6 +275,25 @@ public Principal getPrincipal() {
         return null;
     }
 
+    @Override
+    public Set<Principal> getPrincipals() {
+        Set<Principal> principals = new HashSet<>();
+        Principal primary = getPrincipal();
+        principals.add(primary);
+        principals.add(new RolePrincipal(primary.getName()));
+        for (AttributeStatement attributeStatement : getAttibuteStatements()) {
```

Comment:
Yes, the attributes are built in the code above this when the token is parsed out. It only adds them in if it reads it out of the XML in the first place.